### PR TITLE
Add scaffolding for unimplemented resources plugin.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Follows on from #3581 , just adding the scaffolding (proto file, unimplemented server, Dockerfile update) for the resources plugin from the [design doc](https://docs.google.com/document/d/1z03msZRGsRvcRaom-yrvEwIWcEfNy6fSA5Zg28gjYvA/edit#heading=h.2pgw6nn9xing).

With this small diff:
```diff
--- a/docs/user/manifests/kubeapps-local-dev-values.yaml
+++ b/docs/user/manifests/kubeapps-local-dev-values.yaml
@@ -15,6 +15,11 @@ postgresql:
   existingSecret: postgresql-db
 kubeappsapis:
   replicaCount: 1
+  enabledPlugins:
+    - helm
+    - resources
+  image:
+    tag: dev1
 ingress:
   enabled: true
   hostname: localhost
```

the resources plugin is registered (but methods are unimplemented):

```
grpcurl -plaintext -d '{"installed_package_ref": {"context": {"cluster": "default", "namespace": "test1test2test3test4"}, "plugin": {"name": "helm.packages", "version": "v1alpha1"}, "identifier": "test"}, "watch": true }' -H "Authorization: $token" localhost:8080 kubeappsapis.plugins.resources.v1alpha1.ResourcesService.GetResources
ERROR:
  Code: Unimplemented
  Message: method GetResources not implemented
```

### Benefits

Step 1 towards implementation of resources plugin.

### Applicable issues

- ref #3403

### CODE REVIEW

**Review by gfichtenholt — APPROVED** [review]
looks good to me. Thanks Michael

File: cmd/kubeapps-apis/proto/kubeappsapis/plugins/resources/v1alpha1/resources.proto [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import "google/protobuf/any.proto";
+
+service ResourcesService {
+    rpc GetResources(GetResourcesRequest) returns (stream GetResourcesResponse) {
+        option (google.api.http) = {
+            get: "/plugins/resources/v1alpha1/{installed_package_ref.plugin.name}/{installed_package_ref.plugin.version}/c/{installed_package_ref.context.cluster}/ns/{installed_package_ref.context.namespace}/{installed_package_ref.identifier}"
+        };
+    }
+}
+
```
Comment (antgamdia) [inline cmd/kubeapps-apis/proto/kubeappsapis/plugins/resources/v1alpha1/resources.proto:17]: API docs nitpick hat: I'd rather use this construct to generate both the title and description of each field, just like in [here](https://github.com/kubeapps/kubeapps/blob/v2.4.1/cmd/kubeapps-apis/proto/kubeappsapis/core/packages/v1alpha1/packages.proto#L281-L295):

``` 
// title
//
// long description
whatever whatever whatever whatever 
``` 

Not a big deal though, totally up to you

File: dashboard/package.json [hunk 2]
```diff
@@ -67,6 +67,7 @@
     "redux-devtools-extension": "^2.13.9",
     "redux-thunk": "^2.4.0",
     "remark-gfm": "^3.0.1",
+    "rxjs": "^7.4.0",
```
Comment (antgamdia) [inline dashboard/package.json:70]: Mmm, why are we adding a new dep with this PR? has the ts code generator changed?

**Review by antgamdia — APPROVED** [review]
Great, thanks. A couple of minor things/questions, but ready to go!

File: dashboard/package.json (hunk 2 above)
Comment (absoludity) [reply dashboard/package.json:70]: It seems to be required when adding a streaming grpc endpoint. The generated typescript client for the streaming RPC appears to require it.

File: cmd/kubeapps-apis/proto/kubeappsapis/plugins/resources/v1alpha1/resources.proto (hunk 1 above)
Comment (absoludity) [reply cmd/kubeapps-apis/proto/kubeappsapis/plugins/resources/v1alpha1/resources.proto:17]: Will do, thanks.

**Pull request merged**